ert_shared/ensemble_evaluator/evaluator.py

```diff
@@ -234,20 +237,37 @@ def stop(self):
         self._ws_thread.join()
 
     def run_and_get_successful_realizations(self):
-        try:
-            with self.run() as mon:
-                for _ in mon.track():
-                    pass
-        except ConnectionRefusedError as e:
-            logger.debug(
-                f"run_and_get_successful_realizations caught {e}, cancelling or stopping ensemble..."
-            )
-            if self._ensemble.is_cancellable():
-                self._ensemble.cancel()
-            else:
-                self._stop()
-            self._ws_thread.join()
-            raise
+        monitor_context = self.run()
+        unsuccessfull_connection_attempts = 0
+        while True:
+            try:
+                with monitor_context as mon:
+                    for _ in mon.track():
+                        unsuccessfull_connection_attempts = 0
+                break
+            except ConnectionClosedError as e:
+                logger.debug(
+                    f"Connection closed unexpectedly in run_and_get_successful_realizations: {e}"
+                )
+            except ConnectionRefusedError as e:
+                unsuccessfull_connection_attempts += 1
+                logger.debug(
+                    f"run_and_get_successful_realizations caught {e}."
+                    f"Current unsuccessfull attempts: {unsuccessfull_connection_attempts}"
+                )
+                if (
+                    unsuccessfull_connection_attempts
```

Comment:
`unsuccessfull_connection_attempts` is this for all realizations combined?

Reply:
Not really sure what you mean, we are not connecting to individual realizations? :thinking: This is attempts for that specific evaluation.